packages/kbn-server-route-repository/src/register_routes.ts

```diff
@@ -94,6 +97,10 @@ export function registerRoutes({
       } catch (error) {
         logger.error(error);
 
+        if (isKibanaResponse(error)) {
```

Comment:
this looks odd? are we throwing a kibana response object as if it was an error?

Reply:
I'm also uncertain about it, but let me give you the rationale.

Imagine this handler:
```
({ response, params }) => {
  if (params.someParam === 'some_invalid_value') {
    // For whatever reason we couldn't express this rule in the request validation
    return response.badRequest();
  }
  
  return response.ok('Nice request');
}
```

The inferred response type of this handler is `any`, because `response.badRequest` is typed to `IKibanaResponse<any>` which is extracted to `any`, so that gets combined with `'Nice request'` which just becomes `any` since that is the "larger" type.

If I instead **throw** the `response.badRequest`, then the inferred response goes back to `'Nice request'`.

This feels correct, since the `badRequest` should end up in a catch handler on the client where the type of the error needs to be checked anyway, and the type in the then handler is correctly inferred. 

But it does feel a bit odd to _throw_ a response (even if it's an error response) but at the same time, it all gets translated to a Kibana response before being sent back anyway.

What do you think?

Reply:
Commonly, we use `Boom` for this: e.g. `throw Boom.badRequest()`. It's an actual error object and can easily be converted back into a `KibanaResponse`. We shouldn't throw anything that is not an actual error object, for reasons (no stack trace, no .message etc).

Reply:
I'll cut this out, I think it can be solved by changing the types in the Kibana response factory instead.